h3/src/proto/headers.rs

```diff
@@ -202,6 +201,9 @@ impl Iterator for HeaderIter {
     }
 }
 
+//= https://www.rfc-editor.org/rfc/rfc9114#section-4.3.1
```

Comment:
Looks good so far. Do you mind moving that comment to the `Field::parse` method? The parse method will return the error when `authority` is empty. So one can find the location when looking through the compliance report.
iirc the error comes from here:
https://github.com/hyperium/h3/blob/b44edeb60d234d49c45828395108f7519a048d4b/h3/src/proto/headers.rs#L294